### ISSUE TITLE
`opt-level = s` and `opt-level = z` don't seem to work

### ISSUE THREAD

**Issue by michaelwoerister** [rust-lang/cargo#7240]
**Problem**

Specifying `opt-level = s` or `opt-level = z` for a profile fails with the following error:

```
$ cargo build
error: failed to parse manifest at `/xoxo/stuff/cargo-opt-level-size-bug/Cargo.toml`

Caused by:
  could not parse input as TOML

Caused by:
  invalid number at line 10
```

**Steps**
1. Create a new project with `cargo new repro`
2. Add 
   ```toml
   [profile.dev]
   opt-level = s
   ```
   to `Cargo.toml`.
3. Run `cargo build`

**Possible Solution(s)**

Fix the `Cargo.toml` parser?

**Notes**

Output of `cargo version`:
```
cargo 1.36.0 (c4fcfb725 2019-05-15)
cargo 1.38.0-nightly (26092da33 2019-07-31)
```

**Comment by ehuss** [comment]
They need to be specified as string, like:

```toml
[profile.dev]
opt-level = 's'
```

The TOML language does not allow bare words as values.

**Comment by michaelwoerister** [comment]
Ah, interesting! Considers this a bug report about an unhelpful error message then `:)`

**Comment by aleksator** [comment]
Looks like the error message is coming from https://github.com/alexcrichton/toml-rs
cc @alexcrichton

**Comment by ehuss** [comment]
@aleksator I recently opened https://github.com/alexcrichton/toml-rs/issues/379 regarding the confusing error message here (it has cropped up in a few other places).  I left some very rough guidance there, where `number_or_date` needs to check if the "keylike" value looks like it starts like a number or date, and if not should provide a better error message.

**Comment by aleksator** [comment]
Fixed in toml-rs 0.5.7. We can rename this issue and close it next time we update dependencies.

**Comment by ehuss** [comment]
Yep, just updated via #8772.
